driver-sync/src/test/functional/com/mongodb/client/unified/UnifiedTestSkips.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client.unified;
+
+import com.mongodb.assertions.Assertions;
+
+import static org.junit.jupiter.api.Assumptions.assumeFalse;
+
+public final class UnifiedTestSkips {
+
+    private UnifiedTestSkips() {}
+
+    public static TestDef testDef(final String dir, final String file, final String test) {
+        return new TestDef(dir, file, test);
+    }
+
+    public static final class TestDef {
+        private final String dir;
+        private final String file;
+        private final String test;
+
+        private TestDef(final String dir, final String file, final String test) {
+            this.dir = dir;
+            this.file = file;
+            this.test = test;
+        }
+
+        /**
+         * Test is skipped because it is pending implementation, and there is
+         * a Jira ticket tracking this which has more information.
+         *
+         * @param skip reason for skipping the test; must start with a Jira URL
+         */
+        public Skip skipJira(final String skip) {
+            Assertions.assertTrue(skip.startsWith("https://jira.mongodb.org/browse/JAVA-"));
```

Comment:
Can the skip be due to a DRIVER-XXX ticket? I think checking against `https://jira.mongodb.org/browse/` is enough to make sure a valid JIRA link is provided

Reply:
I considered this, but I think we need to have something our team is tracking. If a good counterexample comes up, we can remove the constraint at the time?